xla/service/gpu/transforms/cudnn_custom_call_compiler.cc

```diff
@@ -541,47 +246,134 @@ absl::StatusOr<se::gpu::CudnnGraph> HloCustomCallToCuDnnGraph(
-        force_deterministic};
-
-    TF_ASSIGN_OR_RETURN(GpufMHABackwardConfig fmha_config,
-                        GpufMHABackwardConfig::For(descriptor));
+    TF_ASSIGN_OR_RETURN(DataType bmm1_grad_gemm1_rhs_type,
+                        GetDNNDataTypeFromPrimitiveType(
+                            bmm1_grad_gemm1_rhs_shape.element_type()));
+    TF_ASSIGN_OR_RETURN(DataType bmm1_grad_gemm2_rhs_type,
+                        GetDNNDataTypeFromPrimitiveType(
+                            bmm1_grad_gemm2_rhs_shape.element_type()));
+    TF_ASSIGN_OR_RETURN(DataType bmm2_grad_gemm1_lhs_type,
+                        GetDNNDataTypeFromPrimitiveType(
+                            bmm2_grad_gemm1_lhs_shape.element_type()));
+    TF_ASSIGN_OR_RETURN(DataType bmm2_grad_gemm2_rhs_type,
+                        GetDNNDataTypeFromPrimitiveType(
+                            bmm2_grad_gemm2_rhs_shape.element_type()));
+    TF_ASSIGN_OR_RETURN(
+        DataType d_output_type,
+        GetDNNDataTypeFromPrimitiveType(d_output_shape.element_type()));
+    TF_ASSIGN_OR_RETURN(
+        DataType d_bmm1_lhs_type,
+        GetDNNDataTypeFromPrimitiveType(d_bmm1_lhs_shape.element_type()));
+    TF_ASSIGN_OR_RETURN(
+        DataType d_bmm1_rhs_type,
+        GetDNNDataTypeFromPrimitiveType(d_bmm1_rhs_shape.element_type()));
+    TF_ASSIGN_OR_RETURN(
+        DataType d_bmm2_rhs_type,
+        GetDNNDataTypeFromPrimitiveType(d_bmm2_rhs_shape.element_type()));
+
+    const DotDimensionNumbers &bmm1_grad_gemm1_dnums =
+        config.bmm1_grad_gemm1_dot_dimension_numbers();
+    const DotDimensionNumbers &bmm1_grad_gemm2_dnums =
+        config.bmm1_grad_gemm2_dot_dimension_numbers();
+    const DotDimensionNumbers &bmm2_grad_gemm1_dnums =
+        config.bmm2_grad_gemm1_dot_dimension_numbers();
+    const DotDimensionNumbers &bmm2_grad_gemm2_dnums =
+        config.bmm2_grad_gemm2_dot_dimension_numbers();
+
+    MatmulTensorDescriptor bmm1_grad_gemm1_rhs = MatmulTensorDescriptor::For(
+        bmm1_grad_gemm1_rhs_type, bmm1_grad_gemm1_rhs_shape.dimensions(),
+        bmm1_grad_gemm1_rhs_shape.layout().minor_to_major(),
+        bmm1_grad_gemm1_dnums.rhs_batch_dimensions(),
+        bmm1_grad_gemm1_dnums.rhs_contracting_dimensions());
+
+    MatmulTensorDescriptor bmm1_grad_gemm2_rhs = MatmulTensorDescriptor::For(
+        bmm1_grad_gemm2_rhs_type, bmm1_grad_gemm2_rhs_shape.dimensions(),
+        bmm1_grad_gemm2_rhs_shape.layout().minor_to_major(),
+        bmm1_grad_gemm2_dnums.rhs_batch_dimensions(),
+        bmm1_grad_gemm2_dnums.rhs_contracting_dimensions());
+
+    MatmulTensorDescriptor bmm2_grad_gemm1_lhs = MatmulTensorDescriptor::For(
+        bmm2_grad_gemm1_lhs_type, bmm2_grad_gemm1_lhs_shape.dimensions(),
+        bmm2_grad_gemm1_lhs_shape.layout().minor_to_major(),
+        bmm2_grad_gemm1_dnums.lhs_batch_dimensions(),
+        bmm2_grad_gemm1_dnums.lhs_contracting_dimensions());
+
+    MatmulTensorDescriptor d_output = MatmulTensorDescriptor::For(
```

Comment:
Nit: maybe this should come after bmm2_grad_gemm2_rhs?

Reply:
Done.